### PR TITLE
RELATED: ONE-4743 Customizable Headline

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
JIRA: ONE-4743

Theme for Headline component
Export Theme contexts for usage in Class components 'static contextType = ThemeContext'

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1639/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3218/ : SUCCESS in 11m 59s

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1650/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/519/ : UNSTABLE in 29m 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1649/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3227/ : SUCCESS in 13m 41s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1651/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3228/ : SUCCESS in 13m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1652/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3229/ : SUCCESS in 14m 13s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by ivan-nejezchleb** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1658/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/521/ : SUCCESS in 28m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1657/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3233/ : SUCCESS in 22m 51s

File: libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx [hunk 1]
```diff
@@ -34,6 +40,11 @@ export interface IThemeProviderProps {
      * component tree.
      */
     workspace?: string;
+
+    /**
+     * If provided it is called with loaded theme to allow its modification according app needs.
```
Comment (no23reason) [inline libs/sdk-ui-theme-provider/src/ThemeProvider/ThemeProvider.tsx:45]: Typo: according to

File: libs/sdk-ui-theme-provider/src/ThemeProvider/Context.tsx [hunk 2]
```diff
@@ -4,10 +4,16 @@ import compose from "lodash/flowRight";
 import { ITheme } from "@gooddata/sdk-backend-spi";
 import { wrapDisplayName } from "@gooddata/sdk-ui";
 
-const ThemeContext = React.createContext<ITheme | undefined>(undefined);
+/**
+ * @beta
+ */
+export const ThemeContext = React.createContext<ITheme | undefined>(undefined);
```
Comment (no23reason) [inline libs/sdk-ui-theme-provider/src/ThemeProvider/Context.tsx:10]: Do we really need to export this? I mean for use in class components you could wrap them in the already present HOCs, right? Or am I missing something?

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/525/ : SUCCESS in 24s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/785/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3245/ : SUCCESS in 13m 34s

**Pull request merged**